src/test/java/com/example/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,244 @@
+        expected.setIsbn("123123123");
+        expected.setPrice(BigDecimal.valueOf(99));
+        expected.setDescription("Great book");
+        expected.setCoverImage("img_src");
+        expected.setCategoriesIds(Set.of(1L));
+
+        MvcResult result = mockMvc.perform(get("/api/books/" + bookId)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class);
+
+        assertNotNull(actual);
+        assertNotNull(actual.getId());
+        Assertions.assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Delete book by valid id")
+    @Sql(scripts = "classpath:database/books/add-book.sql")
+    void deleteBookById_ValidId_Success() {
+        long bookId = 1L;
+
+        mockMvc.perform(delete("/api/books/" + bookId))
+                .andExpect(status().isNoContent())
+                .andReturn();
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Find all books")
+    @Sql(scripts = "classpath:database/books/add-three-books.sql")
+    void findAllBooks_Valid_Success() {
+        BookDto book1 = new BookDto();
+        book1.setId(1L);
+        book1.setTitle("Clean Code");
+        book1.setAuthor("Martin");
+        book1.setIsbn("123123123");
+        book1.setPrice(new BigDecimal("99"));
+        book1.setDescription("Great book");
+        book1.setCoverImage("img_src");
+        book1.setCategoriesIds(Set.of(1L));
+
+        BookDto book2 = new BookDto();
+        book2.setId(2L);
+        book2.setTitle("How to cook");
+        book2.setAuthor("Grandma");
+        book2.setIsbn("777777777");
+        book2.setPrice(new BigDecimal("48"));
+        book2.setDescription("Not bad book");
+        book2.setCoverImage("img_src");
+        book2.setCategoriesIds(Set.of(1L));
+
+        BookDto book3 = new BookDto();
+        book3.setId(3L);
+        book3.setTitle("Ukrainian literature 8 grade");
+        book3.setAuthor("Avramenko");
+        book3.setIsbn("666666666");
+        book3.setPrice(new BigDecimal("10"));
+        book3.setDescription("Awful book");
+        book3.setCoverImage("img_src");
+        book3.setCategoriesIds(Set.of(1L));
+
+        List<BookDto> expected = List.of(book1, book2, book3);
+
+        MvcResult result = mockMvc.perform(get("/api/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String resultAsString = result.getResponse().getContentAsString();
+
+        List<BookDto> actual =
+                objectMapper.readerForListOf(BookDto.class).readValue(resultAsString);
+
+        for (int i = 0; i < expected.size(); i++) {
```

Comment:
I would suggest using assertIterableEquals(expected, actual). In this case you can skip using for loop (but its up to you)